src/edifact.ts

```diff
@@ -1067,25 +1067,34 @@ export class MessageDetails implements Segment {
 
 // DTM
 
-export class DateTimePeriod implements Segment {
-
-    tag = "DTM";
+class DateTimePeriod {
 
     dateTimeOrPeriodFunctionCodeQualifier: string;
     dateTimeOrPeriodText: string | undefined;
     dateTimeOrPeriodFormatCode: string | undefined;
 
-    constructor(data: ResultType) {
-        this.dateTimeOrPeriodFunctionCodeQualifier = data.elements[0][0];
-        if (data.elements[0].length > 1) {
-            this.dateTimeOrPeriodText = data.elements[0][1];
+    constructor(data: string[]) {
+        this.dateTimeOrPeriodFunctionCodeQualifier = data[0];
+        if (data.length > 1) {
+            this.dateTimeOrPeriodText = data[1];
         }
-        if (data.elements[0].length > 2) {
-            this.dateTimeOrPeriodFormatCode = data.elements[0][2];
+        if (data.length > 2) {
+            this.dateTimeOrPeriodFormatCode = data[2];
         }
     }
 }
 
+export class DateTimePeriod implements Segment {
+
+    tag = "DTM";
+
+    dateTimePeriod: DateTimePeriod;
```

Comment:
This here needs to change then as well

```suggestion
    dateTimePeriod: DateTimePeriodData;
```